code/modules/mob/living/carbon/alien/humanoid/queen.dm

```diff
@@ -22,7 +22,20 @@
 	maxHealth = 400
 	health = 400
 	icon_state = "alienq"
+	var/datum/action/small_sprite/smallsprite = new/datum/action/small_sprite()
 
+/datum/action/small_sprite
+	name = "Toggle Giant Sprite - Others will always see you as giant"
+	button_icon_state = "smallqueen"
+	var/small = 0
+
+/datum/action/small_sprite/Trigger()
+	..()
+	if(!small)
+		var/image/I = image('icons/mob/alien.dmi', "alienq_running")
+		owner.add_alt_appearance("alienq_running", I, list(src))

```

Comment:
because when I wrote the example for you, ```src``` was the queen, as it needs to be, you've since moved it to an action toggle (which is a good idea well done) so src is no longer the queen, the queen will be the ```owner``` var.

also this key doesn't need to be ```"alienq_running"``` here, and I'd suggest something like ```"small_queen"```